e2e/tool-panel.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { test, expect } from '@playwright/test';
+import translations from '../client/i18n/locales/english/translations.json';
+
+test.describe('Tool Panel', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(
+      '/learn/javascript-algorithms-and-data-structures/basic-javascript/increment-a-number-with-javascript'
+    );
+  });
+  test('should display "//running tests" in console after clicking "Run the Tests (Ctrl+Enter)" button', async ({
+    page
+  }) => {
+    await page.getByTestId('run-the-tests-btn').click();
+
+    await expect(page.getByTestId('output-text')).toContainText(
+      translations.learn['running-tests']
+    );
+  });
+
+  test('should display reset modal after clicking "Reset this lesson" button', async ({
+    page
+  }) => {
+    await page
+      .getByRole('button', { name: translations.buttons['reset-lesson'] })
+      .click();
+
+    await expect(page.getByTestId('reset-lesson-btn')).toBeVisible();
```

Comment:
We can use the same getByRole method to check for visibility.

Reply:
I decided to check if the 'Reset this lesson?' header is visible because the `button.getByRole('button', { name: translations.buttons['reset-lesson'] })` would correspond to two visible buttons: the button in the modal and the button in the tool panel.